iml-gui/crate/src/components/command_modal.rs

```diff
@@ -572,12 +577,36 @@ fn extract_uri_id<T: EndpointName>(input: &str) -> Option<u32> {
     })
 }
 
-fn is_finished(cmd: &RichCommand) -> bool {
-    cmd.complete
+fn to_load_cmd(model: &Model, cmd_id: u32) -> bool {
+    // load it if the command is not found or is not complete
+    if let Some(cmd) = model.commands.get(&cmd_id) {
+        !cmd.complete
+    } else {
+        true
+    }
+}
+
+fn to_load_job(model: &Model, job_id: u32) -> bool {
+    if let Some(job) = model.jobs.get(&job_id) {
+        // job.state can be "tasked", "pending" or "complete"
+        // if a job is errored or cancelled, it is also complete
+        job.state != "complete"
+    } else {
+        true
+    }
+}
+
+fn to_load_step(model: &Model, step_id: u32) -> bool {
+    if let Some(step) = model.steps.get(&step_id) {
+        // step.state can be "success", "failed" or "incomplete"
+        step.state != "success" && step.state != "failed"
```

Comment:
This will work but if we are comparing state I really think we should be using an enum as opposed to a string.

Reply:
Fair point, I'll create the issue, since making the model more type safe is outside the scope of the task the PR is solving.

Reply:
This can be done (and is a good idea), but it will need to updated in the wire-types since this is referencing an API step.

Reply:
sounds good to me. I'll +1